SimpleImageProcessor/mainwindow.cpp

```diff
@@ -3,12 +3,82 @@
 
 MainWindow::MainWindow(QWidget *parent) :
     QMainWindow(parent),
-    ui(new Ui::MainWindow)
+    ui(new Ui::MainWindow),
+    currScene(new QGraphicsScene),
+    currImg(new QPixmap)
 {
-    ui->setupUi(this);
+    this->ui->setupUi(this);
+    this->ui->graphicsView->setScene(currScene);
 }
 
 MainWindow::~MainWindow()
 {
+    delete currImg;
+    delete currScene;
     delete ui;
 }
+
+void MainWindow::on_actionOpen_triggered()
+{
+    this->File = QFileDialog::getOpenFileName(this,tr("Open Image"),QDir::currentPath(),tr("Image Files (*.jpeg *.jpg *.bmp *.tif *.png)"));
+    this->currImg->load(File);
+    this->currScene->clear();
+    this->currScene->addPixmap(*currImg);
+    this->ui->graphicsView->show();
+}
+
+void MainWindow::on_actionSave_triggered()
+{
+    this->currImg->save(this->File,0,100);
+}
+
+void MainWindow::on_actionSave_As_triggered()
+{
+    QString saveName = QFileDialog::getSaveFileName(this,tr("Save Image"),QDir::currentPath(),tr("Jpeg (*.jpeg);;Bmp (*.bmp);; Tif (*.tif);;Png (*.png)") );
+    this->currImg->save(saveName,0,100);
+}
+
+void MainWindow::on_actionBlur_triggered()
+{
+    /*Apply filter to this->currImg */
+    this->currScene->clear();
+    this->currScene->addPixmap(*currImg);
+    this->ui->graphicsView->show();
+
+}
+
+void MainWindow::on_actionSharpen_triggered()
+{
+    /*Apply filter to this->currImg */
+    this->currScene->clear();
+    this->currScene->addPixmap(*currImg);
```

Comment:
Ne sam ubeden, che towa koeto shte kaja e prawilno. Clear-vaiki scene ti izchistwash pixmap i razni drugi istorii. AddPixmap priema constant reference koeto si e reference to constant i mislq che ne mojesh da prawish promeni po stoinostta kam koqto sochi currImg.